sktime/forecasting/base/adapters/_generalised_statsforecast.py

```diff
@@ -27,10 +31,46 @@ def __init__(self):
         super().__init__()
 
         self._forecaster = None
+        pred_supported = self._check_supports_pred_int()
+        self._support_pred_int_in_sample = pred_supported["int_in_sample"]
+        self._support_pred_int = pred_supported["int"]
 
-    def _instantiate_model(self):
+        self.set_tags(
+            **{"capability:pred_int:insample": self._support_pred_int_in_sample}
+        )
+        self.set_tags(**{"capability:pred_int": self._support_pred_int})
+
+    def _get_statsforecast_class(self):
         raise NotImplementedError("abstract method")
 
+    def _get_statsforecast_params(self):
+        raise NotImplementedError("abstract method")
+
+    def _get_init_statsforecast_params(self):
+        statsforecast_class = self._get_statsforecast_class()
+        return list(signature(statsforecast_class.__init__).parameters.keys())
+
+    def _validate_init_params(self, **sktime_params):
+        sktime_params = sktime_params.copy()
+        statsforecast_params = self._get_init_statsforecast_params()
+
+        for sktime_param in sktime_params.keys():
+            if sktime_param not in statsforecast_params:
+                sktime_params.pop(sktime_param)
+                warn(
```

Comment:
I think the warning should be raised only if the user sets a non-default in `sktime_param`. You can check with via the method `self.get_param_defaults()` which every `skbase` object has.

Reply:
It can also be the case where we have a sktime_param that is not found in the installed version of statsforecast of the user. Not sure what you want to happen if that is the case. So my thought was to skip that parameter and inform the user that this is the case. As they keep changing the API it might be possible that the statsforecast version of the user is too new and maybe some parameters have disappeared eventhough they are in the documentation of sktime.

Reply:
> Not sure what you want to happen if that is the case.

In that case, I would like to:

* warn the user if they pass a non-default
* do not warh the user if they pass a default (e.g., if they do not explicitly set the parameter)

Why: imagine the long-term issue where there is exactly one unique parameter per minor `statsforecast` version, 1.5.0, 1.6.0, 1.7.0, etc, which then disappears again.

In this case, with the current logic, any user would *always* see a warning about the parameters that are no longer even applicable. That seems excessive...

With the proposed logic, they would see warnings only if they actually type out a deprecated parameters (although not "if and only if")